Core/Spells: Increase search radius for avoid problems with some spells that can't find units with huge combat reach

Jildor opened the pull request:
**Changes proposed:**

- Increase search radius for avoid problems with some spells that can't find units with huge combat reach
- Added extra radius to area/cone spells, totems like Searing Totem and SPELL_EFFECT_APPLY_AREA_AURA_ENEMY like Totem of Wrath

Explained in this issue for master branch: https://github.com/TrinityCore/TrinityCore/issues/22375
Creatures like Halion, Sindragosa, Brain of yogg-Saron have big combat reach and if you're are in other cell, you can't find it if search radius is lesser than combat reach

I set a search radius of 40 yards, it seemed enough to me seeing the combat reach of creatures in 335.

I don't know if you think it's okay to do this like this, I can't think of another way to solve it that doesn't cause problems or is within my knowledge

**Issues addressed:**

Closes #27135


**Tests performed:**

tested in-game
before PR:
https://imgur.com/mpLweDJ
https://imgur.com/hYtD6Km

after PR:
https://imgur.com/iWcQ0i9
https://imgur.com/251EGyd

Jildor committed:
Core/Spells: Increase search radius for avoid problems with some spells that can't find units with huge combat reach

Nyr97 commented on src/server/game/Grids/Cells/CellImpl.h:
```diff
@@ -81,6 +81,12 @@ inline void Cell::Visit(CellCoord const& standing_cell, TypeContainerVisitor<T,
     }
+
+    // We need in some cases increase search radius
+    // Allow find creatures with huge combat reach in a diferent nearby cell
```
Spelling.

Faq commented on src/server/game/Grids/Cells/CellImpl.h:
```diff
@@ -81,6 +81,12 @@ inline void Cell::Visit(CellCoord const& standing_cell, TypeContainerVisitor<T,
     }
+
+    // We need in some cases increase search radius
+    // Allow find creatures with huge combat reach in a diferent nearby cell
```
```suggestion
    // Allow find creatures with huge combat reach in a different nearby cell
```

Jildor replied:
thx, done ;)

Jildor committed:
Spelling typo ;)

Shauren commented:
Why are you adding extra argument to `Cell::Visit*` when all it does is modify another of its arguments? Just do it at call site, in spell code

Jildor commented:
You are right, it is simpler.
I'll change it tomorrow, thanks :)

Jildor committed:
apply @Shauren feedback :)

Jildor commented:
Updated :)
I have added a check to add the extra search radius if the default range is greater than 0, is for don't skip this check: https://github.com/TrinityCore/TrinityCore/blob/cb15c9111068f60ed28394cc056c15863ad83813/src/server/game/Grids/Cells/CellImpl.h#L79
If you think that check is not necessary, please tell me and I will remove it

Seyden commented:
Thats not a valid fix if we have creatures with a combat radius which is bigger than 2 cells right? Such as a lot of cataclysm bosses and stuff

Jildor commented:
It is focused on the 3.3.5 branch.
I have no idea of the combat reach of the creatures that are in the master branch, in case there were broader combat reach than in the 3.3.5 branch it would be to increase that value for master, but in 3.3.5 40 seems enough to me.

Shauren commented:
The highest combatreach value in 3.3.5 creature_model_info is 77

Jildor commented:
I know, but I did not see the need to increase the value for a creature that is not worth it, anyway, I can increase it to the value you tell me, I don't have any problem, but as I said at the beginning, it seems enough to me 40

jackpoz reviewed (approved):
Looks good to me, letting Shauren merge it.
Even if there are some edge cases with 77 yards that might still not work, 40 y is good enough of an improvement.
